googlemock/include/gmock/gmock-spec-builders.h

```diff
@@ -1471,7 +1471,8 @@ class FunctionMockerBase : public UntypedFunctionMockerBase {
   virtual ~FunctionMockerBase()
         GTEST_LOCK_EXCLUDED_(g_gmock_mutex) {
     MutexLock l(&g_gmock_mutex);
-    VerifyAndClearExpectationsLocked();
+    if(!std::uncaught_exception())
```

Comment:
Sorry, it is not truth. In C++17 the return type is changed only:
[cppreference.com](http://en.cppreference.com/w/cpp/error/uncaught_exception)

```
bool uncaught_exception();
(1)	(deprecated in C++17)
int uncaught_exceptions();
(2)	(since C++17)
```
(2) Detects how many exceptions in the current thread have been thrown or rethrown and not yet entered their matching catch clauses.

So the proposed change is correct.